### PR TITLE
Hook into Jira provisoning

### PR DESCRIPTION
**Pull request by dagguh**
Depends on #101

### CODE REVIEW

File: src/main/kotlin/com/atlassian/performance/tools/infrastructure/api/jira/install/hook/DataCenterHook.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+) : PostInstallHook {
+
+    override fun call(
+        ssh: SshConnection,
+        jira: InstalledJira,
+        hooks: PostInstallHooks
+    ) {
+        val localSharedHome = sharedHome.localSharedHome
+        sharedHome.mount(ssh)
+        val jiraHome = jira.home.path  // TODO what's the difference between localSharedHome and jiraHome? should both be hookable?
```
Comment (dagguh) [inline src/main/kotlin/com/atlassian/performance/tools/infrastructure/api/jira/install/hook/DataCenterHook.kt:19]: - [ ] remove the DataCenterHook until we get a DC test

File: src/main/kotlin/com/atlassian/performance/tools/infrastructure/api/jira/report/StaticReport.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.atlassian.performance.tools.infrastructure.api.jira.report
+
+import com.atlassian.performance.tools.ssh.api.SshConnection
+
+class StaticReport(
+    private val remotePath: String
+) : Report {
+
+    override fun locate(ssh: SshConnection): List<String> = listOf(remotePath)
```
Comment (dagguh) [inline src/main/kotlin/com/atlassian/performance/tools/infrastructure/api/jira/report/StaticReport.kt:9]: It assumes that the `ssh` will connect to the same server as when it got the `remotePath` during construction.
Might get messy when we get DC hooks support. It'd probably need to return a `List<RemotePath>` and higher layers will have to deal with paths coming from different hosts.

**Review by mzyromski-atlassian — APPROVED** [review]
@dagguh, this work is awesome!

**Comment by dagguh** [comment]
Thanks! ❤️ 
I'm working on Data Center version of this. And it seems the abstractions here are insufficient. We need to distinguish a Jira instance from a Jira node. And an instance home from node home (shared home vs local home). I'll put it back into draft. When this is compatible with DC, I will still reduce the scope of the PR to just Server in order to keep it small enough.

**Comment by dagguh** [comment]
Obsoleted by #103

**Pull request closed without merging**